x/stake/keeper/delegation_test.go

```diff
@@ -581,6 +581,32 @@ func TestRedelegation(t *testing.T) {
 	require.Equal(t, 0, len(redelegations))
 }
 
+func TestRedelegateToSameValidator(t *testing.T) {
+
+	ctx, _, keeper := CreateTestInput(t, false, 0)
+	pool := keeper.GetPool(ctx)
+	pool.LooseTokens = sdk.NewDec(30)
+
+	// create a validator with a self-delegation
+	validator := types.NewValidator(addrVals[0], PKs[0], types.Description{})
+	validator, pool, issuedShares := validator.AddTokensFromDel(pool, sdk.NewInt(10))
+	require.Equal(t, int64(10), issuedShares.RoundInt64())
+	keeper.SetPool(ctx, pool)
+	validator = TestingUpdateValidator(keeper, ctx, validator)
+	pool = keeper.GetPool(ctx)
```

Comment:
I feel like we really need an easier way to setup staking tests... 
This is a lot of overhead for a very simple test.

Reply:
++